### DISCUSSION TITLE
Charm initialization matrix format -- manual alignment with tools like FreeView or ITK-SNAP

### DISCUSSION THREAD

**Discussion by grodgers1** [simnibs/simnibs#172] [7.Creating head models] [answered]
Thank you for the great tool!

I am working with MRI data which has poor initial alignment with the atlas template. I would like to perform a quick manual alignment using a tool like itk-snap or freeview, then use the resulting transformation with charm and the --inittransform option. I am currently struggling to make this work.

So far, I took the transformation matrix from my manual pre-alignment and tried to feed it into charm (converted into a space-delimited 4x4 .txt file), but the results are clearly wrong. It is probably related to the coordinate systems of those tools compared to what is used by charm.

Is there any sort of guide for the matrix to feed into --inittransform? Is there a workflow/tool that fits more seamlessly with simnibs? Any suggestions for how to convert the matrices from .lta from FreeView or the itk-based system from ITK-SNAP?

**Comment by oulap** [comment, marked as answer]
The input matrix should be the world-to-world transformation from the atlas to the input scan in RAS space.

The .lta file should store the type of transform it holds, i.e. ras-to-ras or vox-to-vox, see here:

https://surfer.nmr.mgh.harvard.edu/fswiki/FsTutorial/LtaFormat

The transform type should be ras-to-ras for it to work with charm. I haven't used ITK-SNAP that much, but I presume there's a way to get the ras-to-ras world transform out of it as well.

**Reply by grodgers1** [reply]
Sorry for the late reply! This was helpful, thanks. I just got time to circle back and make this work -- there were a few steps which weren't hard but may not be obvious to others. For future reference, this is how I managed:
- loaded the subject T1 and the template into freeview
- manually aligned them using the transformation tool (Tools->Transform Volume...)
- exported the transformation as an lta (Save Registration...), which produces a vox-to-vox transformation
- converted the vox-to-vox transformation into a ras-to-ras transformation using [lta_convert](https://ftp.nmr.mgh.harvard.edu/pub/docs/html/lta_convert.help.xml.html) -- just use `lta_convert --ltain vox2vox.lta --ltaout ras2ras.lta`
- took the transformation matrix from the resulting .lta and put it alone into a .txt file
- use that .txt for charm's --inittransform argument!
